### DISCUSSION TITLE
IModelApp.settings apis not working

### DISCUSSION THREAD

**Discussion by Rahul-0108** [iTwin/itwinjs-core#2525] [Q&A] [answered]
Hi ,
I am using iTwinViewer. I am trying to save settings in Product Settings Service . The saveuserSetting and getUserSetting apis are working fine.

save SharedSetting api is also working fine . I am getting success response.

![image](https://user-images.githubusercontent.com/53996840/137627999-df89e363-f9e8-4a83-9df5-7bbd11b5838b.png)

and I am saving the setting as like this

![image](https://user-images.githubusercontent.com/53996840/137628108-49cc6c23-4cbe-40c2-b070-2279007c0c7d.png)

the response object while calling saveSharedSetting

{
SettingsResult {status: 0, errorMessage: undefined, setting: undefined}
status:0
setting:undefined
errorMessage:undefined
}

**_Now while trying to fetch the setting using getSharedSetting I am not getting the setting_**

![image](https://user-images.githubusercontent.com/53996840/137628391-77d93a54-e16c-461b-9867-43528020bb17.png)

and this is how I am calling the api

![image](https://user-images.githubusercontent.com/53996840/137628278-cc742934-da13-4849-9fc5-22d223536025.png)

the respone which I am getting is this: 

{
SettingsResult {status: 110597, errorMessage: undefined, setting: undefined}
status:110597
setting:undefined
errorMessage:undefined
}

The application I have created from developer.bentley.com .Can you please tell me what I am doing wrong here ,??

**Comment by pmconne** [comment, marked as answer]
Something is missing from your description. You say the APIs are working fine, and you're getting back a "success" response. So what's the problem?

**Reply by Rahul-0108** [reply]
I have updated the question . I problem is while fetching the setting using getSharedSetting

**Reply by pmconne** [reply]
That error code is "SettingNotFound".
You are saving a shared setting via saveSharedSetting, but querying a user setting via getUserSetting.
If you want to save a user setting, use saveUserSetting. If you want to query a shared setting, use getSharedSetting.

**Reply by Rahul-0108** [reply]
Thanks !!
